utils/generate-commands-json.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+# Generate a commands.json file from the output of the `COMMAND` command
+# Prerequisites: Python 3.5+ and redis-py
+import argparse
+import functools
+import json
+import redis
+
+def convert_list_to_dict(arr, recurse):
+    rep = {}
+    for i in range(0,len(arr),2):
+        key = arr[i].replace('-', '_')
+        val = arr[i+1]
+        if type(val) is list and recurse:
+            val = convert_list_to_dict(val, recurse)
+        rep[key] = val
+    return rep
+
+def convert_flags_to_truthy_dict(flags):
+    rep = {}
+    for f in flags:
+        rep[f] = True
+    return rep
+
+def convert_argument(arg):
+    rep = convert_list_to_dict(arg, False)
+    if 'flags' in rep:
+        rep = {
+            **rep,
+            **convert_flags_to_truthy_dict(rep['flags'])
+        }
+        del rep['flags']
+    if 'arguments' in rep:
+        rep['arguments'] = list(map(lambda x: convert_argument(x), rep['arguments']))
+    return rep
+
+def convert_keyspec(spec):
+    rep = convert_list_to_dict(spec, False)
+    if 'flags' in rep:
+        rep = {
+            **rep,
+            **convert_flags_to_truthy_dict(rep['flags'])
+        }
+        del rep['flags']
+    rep['begin_search'] = convert_list_to_dict(rep['begin_search'], True)
+    rep['find_keys'] = convert_list_to_dict(rep['find_keys'], True)
+    return rep
+
+def convert_command_to_objects(container, cmd):
+    obj = {}
+    rep = [obj]
+    name = cmd[0].upper()
+    key = f'{container} {name}' if container else name
+
+    meta = convert_list_to_dict(cmd[7], False)
+    meta['command_flags'] = cmd[2]
+    meta['acl_categories'] = cmd[6]
+
+    if 'key_specs' in meta:
+        meta['key_specs'] = list(map(lambda x: convert_keyspec(x), meta['key_specs']))
+    if 'arguments' in meta:
+        meta['arguments'] = list(map(lambda x: convert_argument(x), meta['arguments']))
+    if 'subcommands' in meta:
+        sub = list(map(lambda x: convert_command_to_objects(name, x), meta['subcommands']))
+        rep += [s for sc in sub for s in sc]
```

Comment:
... what?

Reply:
flatmap?

Reply:
before this point `sub` is an array of objects, each represents a subcommand, no?
isn't that exactly what you need?